1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,5 +1,58 @@
-function setAlarm() {}
+const bodyEl = document.querySelector("body");
+const timeRemaining = document.getElementById("timeRemaining");
+const alarmSetInput = document.getElementById("alarmSet");
+const pauseAlarmBtn = document.getElementById("pause");
+//pauseAlarmBtn.innerHTML = "Pause Alarm";
+let BgInterval;
+let timerInterval;
 
+function setAlarm() {
+  let setTime = alarmSetInput.value;
+
+  timeRemaining.innerText = formatSetTime(setTime);
+  bodyEl.style.backgroundColor = "rgb(256,256,256)";
+  clearInterval(BgInterval);
+
+  timerInterval = setInterval(() => {
+    setTime = setTime - 1;
+    timeRemaining.innerText = formatSetTime(setTime);
+
+    if (setTime === 0) {
+      clearInterval(timerInterval);
+      setBackgroundColor();
+      playAlarm();
+    }
+  }, 1000);
+}
+
+function setBackgroundColor() {
+  BgInterval = setInterval(() => {
+    var x = Math.floor(Math.random() * 256);
+    var y = Math.floor(Math.random() * 256);
+    var z = Math.floor(Math.random() * 256);
+    bodyEl.style.backgroundColor = "rgb(" + x + "," + y + "," + z + ")";
+  }, 1000);
+}
+
+function formatSetTime(timer) {
+  const second = timer % 60;
+  const minute = Math.floor(timer / 60);
+  return `Time Remaining: ${minute > 9 ? minute : "0" + minute}:${
+    second > 9 ? second : "0" + second
+  }`;
+}
+
+// function pauseAlarm() {
```

Comment:
You were so close with this one Laleh! 

The big issue I can see here is that `setTime` is bound to the `setAlarm` function scope. So when you try to continue in the `pause` function, the code has no idea what the current time is. 

What I did was move the `setTime` outside of the `setTime` function as a global variable (we call this `state`, but more of that when we move to React!). Then, when you run the pause function and need time to continue, the code has a place where it can get that piece of state.

Here is what I did:

```
let time;

function setAlarm() {
	time = alarmSetInput.value ? alarmSetInput.value : time;
	alarmSetInput.value = null;

	timeRemaining.innerText = formatSetTime(time);
	bodyEl.style.backgroundColor = "rgb(256,256,256)";
	clearInterval(BgInterval);

	timerInterval = setInterval(() => {
		time = time - 1;
		timeRemaining.innerText = formatSetTime(time);

		if(time === 0) {
			clearInterval(timerInterval);
			setBackgroundColor();
			playAlarm();
		}
	}, 1000);
}

function pauseAlarm() {
	buttonType == "Pause Alarm" ? ()=> { buttonType = "Continue"; clearInterval(timerInterval); }
		:
		buttonType = "Pause Alarm";
}

function setup() {
	document.getElementById("set").addEventListener("click", () => {
		setAlarm();
	});

	document.getElementById("stop").addEventListener("click", () => {
		stopAlarm();
	});

	pauseAlarmBtn.addEventListener("click", () => {
	  buttonType == "Pause Alarm" ? pauseAlarm() : setAlarm();
	});
}
```

Great job otherwise! As always, your code works really well and is very easy to read!

Reply:
Oh, I also added  `<button id="pause" type="button">Pause Alarm</button>` to the html :)